src/libpmemset/part.c

```diff
@@ -5,17 +5,55 @@
  * part.c -- implementation of common part API
  */
 
-#include "part.h"
+#include <fcntl.h>
+
 #include "libpmemset.h"
+#include "os.h"
+#include "part.h"
+#include "pmemset.h"
+#include "pmemset_utils.h"
+#include "source.h"
+
+struct pmemset_part {
+	struct pmemset *set;
+	struct pmemset_source *src;
+	size_t offset;
+	size_t length;
+};
 
 /*
- * pmemset_part_descriptor_new -- not supported
+ * pmemset_part_new -- creates a new part for the provided set
  */
 int
 pmemset_part_new(struct pmemset_part **part, struct pmemset *set,
 		struct pmemset_source *src, size_t offset, size_t length)
 {
-	return PMEMSET_E_NOSUPP;
+	LOG(3, "part %p set %p src %p offset %zu length %zu",
+			part, set, src, offset, length);
+	PMEMSET_ERR_CLR();
+
+	int ret;
+	struct pmemset_part *partp;
+	*part = NULL;
+
+	ret = pmemset_source_validate(src);
+	if (ret)
+		return ret;
+
+	partp = pmemset_malloc(sizeof(*partp), &ret);
+	if (ret)
+		return ret;
+
+	ASSERTne(partp, NULL);
+
+	partp->set = set;
```

Comment:
this function should open file (or extract handle/fd), and store handle/fd in pmemset_part_structure